src/pages/sign-on/components/desktop/SignInPage.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState, useEffect, useCallback, KeyboardEvent } from 'react'
+
+import { Button, ButtonType, IconArrow } from '@audius/stems'
+import cn from 'classnames'
+import { Spring } from 'react-spring/renderprops'
+
+import audiusLogoColored from 'assets/img/audiusLogoColored.png'
+import Input from 'components/data-entry/Input'
+import LoadingSpinner from 'components/loading-spinner/LoadingSpinner'
+import PreloadImage from 'components/preload-image/PreloadImage'
+import StatusMessage from 'components/status-message/StatusMessage'
+
+import { ForgotPasswordHelper } from './ForgotPasswordHelper'
+import { MetaMaskOption } from './MetaMaskOption'
+import styles from './SignInPage.module.css'
+
+const messages = {
+  title: 'Sign Into Your Audius Account',
+  error: {
+    inUse: 'Invalid password',
+    default: 'Invalid Credentials'
+  },
+  forgotPasswordText: 'Forgot your password?'
+}
+
+type emailStatus = 'editing' | 'failure' | 'loading' | 'success'
+type passwordStatus = 'editing' | 'failure' | 'loading'
+
+type SignInProps = {
+  isMobile: boolean
+  loading: boolean
+  hasMetaMask: boolean
+  email: {
+    value: string
+    error: string
+    status: emailStatus
+  }
+  password: {
+    value: string
+    error: string
+    status: passwordStatus
+  }
+  onEmailChange: (email: string, validate?: boolean) => void
+  onPasswordChange: (password: string) => void
+  onSignIn: (email: string, password: string) => void
+  onSignUp: () => void
+  onMetaMaskSignIn: () => void
+}
+
+export const SignInPage = ({
+  isMobile,
+  loading,
+  hasMetaMask,
+  email,
+  password,
+  onPasswordChange,
+  onEmailChange,
+  onSignIn,
+  onSignUp,
+  onMetaMaskSignIn
+}: SignInProps) => {
+  const [showForgotPassword, setShowForgotPassword] = useState(false)
+
+  const passwordInput = React.createRef<HTMLInputElement>()
+
+  useEffect(() => {
+    if (email && email.value) {
+      passwordInput.current?.focus()
+    }
+    // We only want to check on first component render (i.e componentDidMount)
```

Comment:
passwordInput can go in the deps here (the ref only changes once at mount)

as for email & email.value, email might be able to be in the deps here because it's an object whose reference doesn't change, but if that's the case where it does change, we should probably wrap this in a ref (we have a nice useInstanceVar hook) that will do what you want. In other words

```
const [getInitialEmail, _] = useInstanceVar(email)

useEffect(() => {
  const email = getInitialEmail()
  if (email && email.value) ...
}, [getInitialEmail, passwordInput])
```